### ISSUE TITLE
Error in `safenormalize` when used with multiple vectors

### ISSUE THREAD

**Issue by dpanici** [PlasmaControl/DESC#1743]
```python
import numpy as np
from desc.utils import safenormalize

a = np.array([[1,2,3],[4,5,6],[7,8,9]])
a_norm = a / np.linalg.norm(a,axis=-1)[:,None]
a_safenorm = safenormalize(a,axis=1)

print(np.linalg.norm(a_norm,axis=1))
print(np.allclose(np.linalg.norm(a_norm,axis=1), 1))
print(np.linalg.norm(a_safenorm,axis=1))
print(np.allclose(np.linalg.norm(a_safenorm,axis=1), 1))
```

```
[1. 1. 1.]
True
[0.41202958 1.28572916 2.17914997]
False
```

I think the culprit is [this line](https://github.com/PlasmaControl/DESC/blob/ed506fa5b3a78e6fcfdc1d24a88dc4f5f0266c4b/desc/utils.py#L939)

```python
n = safenorm(x, ord, axis, fill, threshold) * jnp.ones_like(x)
```

Which is the transpose of what we actually want (we want an array of the same size as `x` but with the norm of the vector in each place, so that n[0,:] is the norm of x[0,:]) i.e. currentlt for the above `n` is

```
array([[ 3.74165739,  8.77496439, 13.92838828],
       [ 3.74165739,  8.77496439, 13.92838828],
       [ 3.74165739,  8.77496439, 13.92838828]])
```

when it really should be 

```
array([[ 3.74165739,  3.74165739,  3.74165739],
       [ 8.77496439,  8.77496439,  8.77496439],
       [13.92838828, 13.92838828, 13.92838828]])
```

**Comment by dpanici** [comment]
This is not an issue when used with a single vector, which in the current code we only ever use `safenormalize` in the `compute/_curve.py` where we only use it on the `normal` vector which is a single vector, and inside `geom_utils.py` where it is used on `axis` which again is a single vector. So this bug is never seen in the current master code
